crates/proof_of_stake/src/queries.rs

```diff
@@ -59,26 +85,55 @@ pub fn find_delegations<S>(
 where
     S: StorageRead,
 {
-    let bonds_prefix = storage_key::bonds_for_source_prefix(owner);
-    let params = read_pos_params(storage)?;
-    let mut delegations: HashMap<Address, token::Amount> = HashMap::new();
-
-    for iter_result in
-        namada_storage::iter_prefix_bytes(storage, &bonds_prefix)?
-    {
-        let (key, _bond_bytes) = iter_result?;
-        let validator_address = storage_key::get_validator_address_from_bond(
-            &key,
-        )
-        .ok_or_else(|| {
-            namada_storage::Error::new_const(
-                "Delegation key should contain validator address.",
-            )
-        })?;
-        let deltas_sum = bond_handle(owner, &validator_address)
-            .get_sum(storage, *epoch, &params)?
-            .unwrap_or_default();
-        delegations.insert(validator_address, deltas_sum);
+    let validators = delegation_targets_handle(owner);
+    if validators.is_empty(storage)? {
+        return Ok(HashMap::new());
+    }
+
+    let mut delegations = HashMap::<Address, token::Amount>::new();
+
+    for validator in validators.iter(storage)? {
+        let (
+            val,
+            DelegationEpochs {
+                prev_ranges,
+                last_range: (last_start, last_end),
+            },
+        ) = validator?;
+
+        // TODO: Should use raw bonds or slashed stake??
```

Comment:
I think raw is ok for its use-cases

Reply:
Right now it is applying slashes, do you think that is ok too?

Reply:
Fixed to just use unslashed amounts in e174eadf0f726cab39703ebd232a487349bb7262